### ISSUE TITLE
bug: tmux with 'sensibleOnTop' option no longer uses the correct shell

### ISSUE THREAD

**Issue by cixel** [nix-community/home-manager#5952]
### Are you following the right branch?

- [X] My Nixpkgs and Home Manager versions are in sync

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Issue description

The upstream version of `tmux` was updated last week to 3.5a ([commit](https://github.com/NixOS/nixpkgs/commit/7facbffaf151f76629f1d67a67d109232ddb0b52)). One of the [changes](https://github.com/tmux/tmux/blob/tmux-3.5a/CHANGES#L7) in this version (from 3.5) is that `/bin/sh` is used for run-shell.

With this version of tmux, all panes/windows/etc use `sh` instead of the configured default shell (in my case, `zsh`).

My rough guess is that this change in tmux has introduced some ordering issues when `sensibleOnTop` is set, as `run-shell` will now use `sh` regardless of how tmux's default-shell is set.

The problem goes away if I do:
```
  package = pkgs.tmux.overrideAttrs (old: rec {
    version = "3.5";
    src = pkgs.fetchFromGitHub {
      owner = "tmux";
      repo = "tmux";
      rev = version;
      hash = "sha256-8CRZj7UyBhuB5QO27Y+tHG62S/eGxPOHWrwvh1aBqq0=";
    };
  });
```

or if I set `sensibleOnTop = false;`

### Maintainer CC

@jorsn @rycee (cc based on https://github.com/nix-community/home-manager/commit/a7cdfaa32585fce404cf8890bae099348e53e0ad).

### System information

```markdown
- system: `"aarch64-darwin"`
 - host os: `Darwin 23.6.0, macOS 14.7`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.18.8`
 - channels(ehdens): `"home-manager, nixpkgs"`
 - nixpkgs: `/Users/ehdens/.nix-defexpr/channels/nixpkgs`
```
```

**Comment by x123** [comment]
I'm also experiencing the same issue. The notes for tmux 3.5a here https://github.com/tmux/tmux/issues/4162 say the following:

"Note that a fix in 3.5 and 3.5a to correctly set SHELL for run-shell and if-shell reveals a bug in the tmux-sensible script, causing it to set default-shell to /bin/sh. This can be worked around by adding this to .tmux.conf after the tmux-sensible script is invoked:"

**Comment by gshpychka** [comment]
Same here. Having `set -g default-shell "$SHELL"` as the last line in my `tmux.conf` does not fix the issue, only disabling the sensible plugin by setting `sensibleOnTop` to `false` does.

**Comment by YorikSar** [comment]
Quick and (very) dirty workaround for this:

```
  nixpkgs.overlays = [
    (final: prev: {
      tmuxPlugins = prev.tmuxPlugins // {
        sensible = prev.tmuxPlugins.sensible.overrideAttrs (prev: {
          postInstall = prev.postInstall + ''
            sed -e 's:\$SHELL:/bin/zsh:g' -i $target/sensible.tmux
          '';
        });
      };
    })
  ];
```

It hardcodes `/bin/zsh` (replace it with what you like to use) and rebuilds tmux-sensible with it.

**Comment by art1es23** [comment]
> I'm also experiencing the same issue. The notes for tmux 3.5a here [tmux/tmux#4162](https://github.com/tmux/tmux/issues/4162) say the following:
> 
> "Note that a fix in 3.5 and 3.5a to correctly set SHELL for run-shell and if-shell reveals a bug in the tmux-sensible script, causing it to set default-shell to /bin/sh. This can be worked around by adding this to .tmux.conf after the tmux-sensible script is invoked:"
> 
> ```
>  - system: `"aarch64-darwin"`
>  - host os: `Darwin 24.0.0, macOS 15.0.1`
>  - multi-user?: `yes`
>  - sandbox: `no`
>  - version: `nix-env (Nix) 2.18.8`
>  - channels(root): `"nixpkgs"`
>  - nixpkgs: `/nix/store/60sn02zhawl3kwn0r515zff3h6hg6ydz-source`
> ```
> 
> EDIT: Using the following in `extraConfig` to make sure it's run after `tmux-sensible` loads seems to fix the issue on darwin and is the updated recommendation from [tmux/tmux#4162 (comment)](https://github.com/tmux/tmux/issues/4162#issuecomment-2412201969)
> 
> ```
> set -gu default-command
> set -g default-shell "$SHELL"
> ```

This solved the root problem, but after running tmux I got another issue: 
`/Users/qmpwwsd/.config/tmux/tmux.conf:91: not a suitable shell: zsh `

After pressing the 'Enter' key: 
- if this is a new session: zsh is used
- if the session is saved: all panels use bash by default, not zsh, only after closing the panel and opening it again zsh is used

And I don't know how to fix it

**Comment by zoriya** [comment]
`SHELL` needs to be an absolute path, not a binary name. You'd need to set it to something like `/run/current-system/sw/bin/zsh`

**Comment by gen740** [comment]
I'm encouraging the same issue and at that time I've noticed the home-manager's tmux automatically load the sensible-plugin. I love plentiful options, but it is necessary to load this plugin by default?

**Comment by gshpychka** [comment]
> I encountered the same issue, and at that time, I noticed that home-manager's tmux automatically loads the sensible plugin. I love having plentiful options, but is it necessary to load this plugin by default?

It's supposed to be a sensible default, hence the name ;)

**Comment by gen740** [comment]
> It’s supposed to be a sensible default, hence the name ;)

This is simply the origin of the plugin’s name. Regardless of the name, it’s still just a plugin.

Who would expect that setting tmux.enable = true would also load an additional tmux plugin?

**Comment by gen740** [comment]
And unfortunately, it seems that tmux-sensible is no longer being maintained😢 

https://github.com/tmux-plugins/tmux-sensible/pull/70#issuecomment-2127666779

**Comment by gshpychka** [comment]
> > It’s supposed to be a sensible default, hence the name ;)
> 
> This is simply the origin of the plugin’s name. Regardless of the name, it’s still just a plugin.
> 
> Who would expect that setting tmux.enable = true would also load an additional tmux plugin?

I don't disagree, was mostly joking.

Yeah, considering the last commit was over 2 years ago - could be worth considering changing the default.

**Comment by gen740** [comment]
So is it OK to create the PR on this?
Or is more discussion needed?

**Comment by gen740** [comment]
So are there any problems in creating the PR in this?

I think a bit more discussion is needed, but it is inconvenient to discuss in this thread.

**Comment by fersilva16** [comment]
Good to note that the default-command `reattach-to-user-namespace` is no longer needed for Tmux 2.6+

https://github.com/ChrisJohnsen/tmux-MacOSX-pasteboard/issues/66
